docs/bundler/hmr.md

```diff
@@ -0,0 +1,266 @@
+
+## Usage
+
+There are two ways to get started with `import.meta.hot` in Bun:
+
+### Frontend Hot Reloading
+
+Hot Reloading is enabled by default when you run `bun ./index.html`.
+
+From there, you can use `import.meta.hot.data` to persist state between module evaluations. The `import.meta.hot` object is referentially equal between module evaluations, so you can use it to persist state across module updates.
+
+To disable HMR when using HTML entrypoints, set `NODE_ENV=production` as an environment variable.
+
+### Full-Stack Hot Reloading
+
+Frontend and server-side hot reloading are independent features that can be used together.
+
+To enable frontend hot reloading, pass `development: true` to `Bun.serve()`.
+
+```ts#server.ts
+import { serve } from "bun";
+import homepage from "./index.html";
+
+serve({
+  // Enable frontend hot reloading
+  development: true,
+
+  routes: {
+    // Setup frontend routes
+    "/": homepage,
+
+    // Setup API routes
+    "/api/v1/users": {
+      async GET() {
+        return Response.json([{ id: 1, name: "John Doe" }]);
+      },
+
+      async POST(req) {
+        const user = await req.json();
+        return Response.json({ success: true, user });
+      },
+    },
+  },
+});
+```
+
+To enable server-side hot reloading, pass `--hot` to the Bun CLI:
+
+```sh
+$ bun --hot ./server.ts
+```
+
+`--hot` and `development: true` can both be used at the same time.
+
+## API Reference
+
+### Compatibility Table
+
+|     | Method             | Notes                                                                             |
+| --- | ------------------ | --------------------------------------------------------------------------------- |
+| 🚧  | `hot.accept()`     | API exists but only logs "TODO" warning; accepts no module boundaries             |
+| ✅  | `hot.dispose()`    | Fully functional; registers callbacks that are executed before module replacement |
+| ✅  | `hot.data`         | Fully functional; persists data between module evaluations                        |
+| ❌  | `hot.decline()`    | Exists as empty function for compatibility with Vite APIs                         |
+| 🚧  | `hot.invalidate()` | Exists but throws "TODO: implement ImportMetaHot.invalidate" error                |
+| 🚧  | `hot.on()`         | Exists but throws "TODO: implement ImportMetaHot.on" error                        |
+| 🚧  | `hot.off()`        | Exists but throws "TODO: implement ImportMetaHot.off" error                       |
+| 🚧  | `hot.send()`       | Exists but throws "TODO: implement ImportMetaHot.send" error                      |
+| 🚧  | `hot.prune()`      | Exists but throws "TODO: implement ImportMetaHot.prune" error                     |
+
+**Currently Functional APIs:**
+
+- `hot.dispose()`: Successfully registers cleanup callbacks that run before module replacement
+- `hot.data`: Successfully maintains state between module evaluations
+
+### `hot.accept()`
```

Comment:
i think for the APIs which are not implemented, we should comment out all this markdown. we can uncomment once it is implemented. the bun docs should document what we have and not a bunch of docs for stuff that doesn't work